### PR TITLE
[Ready for Review] User can see their charts in the report tab

### PR DESCRIPTION
**Pull request by katchen1**
# Overview
This allows the user to view a report based on their watch history by clicking the "Report" tab. I used the [MPAndroid Chart library](https://github.com/PhilJay/MPAndroidChart) to create the charts. Specifically, I used Bar Charts (horizontal and stacked), a Pie Chart, and a Line Chart. 

The report shows the following:

* Overview Section
  * Total number of animes watched
  * Average rating
  * Number of animes watched this year (+comparison to previous year)
  * Average rating this year (+comparison to previous year)
* Activity (Line Chart of year versus anime count)
* Genre Distribution (Pie Chart of genre versus anime count)
* Genre Preference (Bar Chart of genre versus average rating of animes with the genre)
* Genre Breakdown by Year (Stacked Bar Chart of year versus anime count, with each bar broken down by genre)
* The user's top 5 anime (shown as entry cards and can click to view the entry's details)
* If the user clicks a data point, a [Marker View](https://github.com/PhilJay/MPAndroidChart/wiki/MarkerView) pops up and they can see the detailed stats (x and y values)
* All charts are animated with cool effects

# Screen Recording
report

### CODE REVIEW

File: app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java [hunk 1]
```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
...
+            }
+        }
+    }
+
+    /* Activity (Line Chart) - year vs. entry count */
+    public void setChartActivity() {
+        LineChart chart = binding.chartActivity;
+
+        // Create data points
+        List<com.github.mikephil.charting.data.Entry> chartEntries = new ArrayList<>();
```
Comment (annechen112995) [inline app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java:177]: nit: can you import this directly above and use `List<Entry>` here instead?

File: app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java [hunk 2]
```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
...
+        List<Integer> entryCounts = new ArrayList<>();
+        for (int year = minYear; year <= maxYear; year++) {
+            int entryCount = yearToList.containsKey(year)? yearToList.get(year).size(): 0;
+            chartEntries.add(new com.github.mikephil.charting.data.Entry(year, entryCount));
```
Comment (annechen112995) [inline app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java:181]: ^

File: app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java [hunk 3]
```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
...
+        dataSet.setDrawValues(false);
+        chart.setData(new LineData(dataSet));
+
+        // Customize the chart
+        customizeChart(chart, 0);
+        customizeXAxis(chart.getXAxis());
+        customizeYAxisMain(chart.getAxisLeft());
+        customizeYAxisSecondary(chart.getAxisRight());
+        chart.getXAxis().setAxisMinimum(chartEntries.get(0).getX());
+        chart.getAxisLeft().setLabelCount(Collections.max(entryCounts) + 2, true);
```
Comment (annechen112995) [inline app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java]: `chart.getAxisLeft()` seems to be called a few times here. Would it make sense to pull this into a separate variable?

File: app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java [hunk 4]
```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
...
+
+    /* General styling of a chart. Mode is for different ways of drawing the marker view. */
+    public void customizeChart(Chart chart, int mode) {
+        chart.getDescription().setEnabled(false);
+        chart.getLegend().setTextColor(white);
+        chart.getLegend().setWordWrapEnabled(true);
+        chart.getLegend().setYOffset(3f);
+        CustomMarkerView mv = new CustomMarkerView(getContext(), R.layout.custom_marker_view_layout, mode);
+        mv.setChartView(chart);
+        chart.setMarker(mv);
```
Comment (annechen112995) [inline app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java:371]: nit: This file is getting to be quite long. I know you have a lot of helper functions here. Is there any way to break this file down a bit more?

**Review by annechen112995 — APPROVED** [review]

File: app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java (hunk 1 above)
Comment (katchen1) [reply app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java:177]: I tried doing that, but it coincides with my own class named `Entry`, and it seems like Java doesn't have an import aliasing mechanism.

**Pull request merged**